projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -723,11 +723,16 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked, OnDestro
   private getFilteredCaseEventData(caseFields: CaseField[], formValue: object, clearEmpty = false,
     clearNonCase = false, fromPreviousPage = false): CaseEventData {
     // Get the data for the fields specified.
-    const formFields = this.formValueService.filterCurrentPageFields(caseFields, formValue);
+    let formFields = this.formValueService.filterCurrentPageFields(caseFields, formValue);
 
     // Sort out the dynamic lists.
     this.formValueService.sanitiseDynamicLists(caseFields, formFields);
 
+    // we only want to ensure this when the field_type is a collection of complex fields
+    if (caseFields.find((field) => field.field_type?.type === 'Collection' && field.field_type?.collection_field_type?.type === 'Complex')) {
```

Comment:
Not hugely significant, but logically you want to use some() here, not find(). I'm a bit confused as to why you check for complex fields - it doesn't look as though the code will do anything with complex